server/src/main/java/org/apache/uniffle/server/HugePartitionUtils.java

```diff
@@ -148,4 +148,9 @@ public static boolean limitHugePartition(
     }
     return false;
   }
+
+  public static boolean hasExceedPartitionSplitLimit(
```

Comment:
Why not extract the general logic to limit the partitioned data? I think the partition split and huge partition limit could be scoped in the same abstract limit policies.

Reply:
Address this after other comments addressed and tested.